src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -15,8 +16,9 @@ public interface MachineService {
      *
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
+     *
      */
-    List<Object> getAll(Class type);
+    List<Machine> getAll(Class type);
```

Comment:
in each method in this interface, use T class type parameter, which you defined above, instead. What was the point of having T if not using it ;)